### PR TITLE
CDAP-1759: Namespace TestManager

### PR DESCRIPTION
**Pull request by jwang47**
https://issues.cask.co/browse/CDAP-1759
http://builds.cask.co/browse/CDAP-RBT135

### CODE REVIEW

**Comment by jwang47** [comment]
@bdmogal Regarding your comment "At least in the program lifecycle APIs, I know there is a jira to drop support for procedures with v3 apis. we may want to remove this/ always use default namespace", it's difficult to do this at the moment since most of the tests in `TestFrameworkTest` are using procedures to check results. I suggest we create another issue to remind us to port `TestFrameworkTest` to use services instead later.

**Comment by bdmogal** [comment]
That's fine then. Let's do it as a separate jira/PR

File: cdap-proto/src/main/java/co/cask/cdap/proto/Id.java [hunk 1]
```diff
@@ -240,6 +244,67 @@ public String toString() {
   }
 
   /**
+   * Uniquely identifies a Schedule.
+   */
+  public static class Schedule {
```
Comment (anwar6953) [inline cdap-proto/src/main/java/co/cask/cdap/proto/Id.java]: unused?

File: cdap-test/src/main/java/co/cask/cdap/test/ApplicationManager.java [hunk 2]
```diff
@@ -101,8 +101,11 @@
    * @param dataSetName Name of the dataset to retrieve.
    * @param <T> Type of the dataset.
    * @return A {@link DataSetManager} instance.
+   * @deprecated As of version 2.8.0, replaced by
+   *             {@link TestManager#getDataset(co.cask.cdap.proto.Id.Namespace, String)}
```
Comment (anwar6953) [inline cdap-test/src/main/java/co/cask/cdap/test/ApplicationManager.java]: I think this is wrong.
It's replaced by TestBase#getDataset, right?

**Comment by anwar6953** [comment]
remainder lgtm

**Comment by anwar6953** [comment]
:+1:

**Pull request merged**